### DISCUSSION TITLE
PyBullet URDF use self collision exclude parent not working

### DISCUSSION THREAD

**Discussion by ARoefer** [bulletphysics/bullet3#3572] [General]
Hello,

I noticed that the URDF_USE_SELF_COLLISION_EXCLUDE_PARENT flag doesn't seem to be working anymore in the current version of PyBullet. URDF_USE_SELF_COLLISION seems to work though. 

Here is a photo showing two KUKAs initialized with the same joint configurations using the different flags. The configuration puts them in a self-intersecting configuration, which gets resolved for the one using the regular self collision flag (right) but not the other one.
![broken_self_collision_exclude_parent](https://user-images.githubusercontent.com/16648305/41157204-2a5be56a-6b25-11e8-8e80-38294043cfac.png)

Code to recreate the problem:
[self_collision_exclude_parent_bug.zip](https://github.com/bulletphysics/bullet3/files/2084270/self_collision_exclude_parent_bug.zip)

**Comment by erwincoumans** [comment]
Need to look into it. Unfortunately the project lacks resources, so it can take a while.

**Comment by erwincoumans** [comment]
Until we get to this, you can achieve the same (and more fine-grained) using collision filters.
See 'setCollisionFilterPair' in the [PyBullet Quickstart Guide](https://docs.google.com/document/d/10sXEhzFRSnvFcl3XxNGhnD4N2SedqwdAvK3dsihxVUA/edit#).
